### PR TITLE
chore: fix typo & add spell check

### PR DESCRIPTION
**Pull request by spacewander**

### CODE REVIEW

**Comment by AsterDY** [comment]
Pls submit more as much as possible

**Comment by spacewander** [comment]
@AsterDY 
Would you like to check the spell in the CI like https://github.com/apache/apisix/blob/a0b1a4554dc9ce23e877cc1fc58507c795203878/.github/workflows/lint.yml#L15?

**Comment by AsterDY** [comment]
sure，you can add it

**Comment by spacewander** [comment]
@AsterDY 
Updated. Please take a look if you have free time.

**Comment by spacewander** [comment]
It seems the self-hosted runner doesn't have pip installed. I will handle it later.

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    runs-on: [self-hosted, X64]
+    steps:
+      - uses: actions/checkout@v4
+      - name: spell check
+        run: |
+          # the python version in self-hosted runner is 3.7, which is not supported by the latest get-pip.py
+          python3 <(curl -s https://bootstrap.pypa.io/pip/3.7/get-pip.py)
+          pip install codespell==2.3.0
+          # ignore test files, go project names, binary files via `--skip` and special var/regex via `--ignore-words`
+          git grep --cached -l '' | xargs codespell  --skip 'fuzz/*,*_test.tmpl,testdata/*,*_test.go,go.mod,go.sum,*.gz' --ignore-words=.github/workflows/.ignore_words
```
Comment (ccoVeille) [inline .github/workflows/lint.yml:16]: I would simply use 

https://github.com/codespell-project/actions-codespell

So this


```suggestion
    steps:
      - uses: actions/checkout@v4
      - name: spell check
      - uses: codespell-project/actions-codespell@v2
```

 So with the .codespellrc file I'm suggesting in my other comment

**Review by ccoVeille — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yml (hunk 1 above)
Comment (spacewander) [reply .github/workflows/lint.yml:16]: @ccoVeille 
I would like to apply your suggestion, but there is a syntax error in it. Would you update it so I can commit the suggestion? Thanks!

Comment (ccoVeille) [reply .github/workflows/lint.yml:16]: oh indeed

Comment (ccoVeille) [reply .github/workflows/lint.yml:16]: is it OK now ?

Comment (AsterDY) [reply .github/workflows/lint.yml:16]: ok

Comment (ccoVeille) [reply .github/workflows/lint.yml:16]: @spacewander ?

**Review by ccoVeille — APPROVED** [review]

**Comment by AsterDY** [comment]
ci有问题，要不直接用github官方镜像跑吧，不要用“self-host” @spacewander

**Comment by spacewander** [comment]
@AsterDY 
Lint can pass now: https://github.com/spacewander/sonic/actions/runs/12517330733/job/34917916275?pr=1

File: internal/decoder/api/decoder_arm64.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ var (
 
 
 func init() {
-    // whe in aarch64. we enable all optimize
+    // when in aarch64, we enable all optimization
```
Comment (ccoVeille) [inline internal/decoder/api/decoder_arm64.go:33]:
```suggestion
    // when in arch64, we enable all optimization
```
Typo?

But shouldn't it be arm64?

**Review by ccoVeille — APPROVED** [review]

File: internal/decoder/api/decoder_arm64.go (hunk 2 above)
Comment (spacewander) [reply internal/decoder/api/decoder_arm64.go:33]: aarch64 is the alias of arm64.

**Review by AsterDY — APPROVED** [review]

**Review by AsterDY — APPROVED** [review]

**Pull request merged**